File: DotnetApi/Services/Impl/QuotesService.cs
using DotnetApi.DB;
using DotnetApi.Entities;

namespace DotnetApi.Services.Impl;

public class QuotesService : IQuotesService
{
    private readonly IQuotesRepository quotesRepository;
    private readonly Random random = new();

    public QuotesService(IQuotesRepository quotesRepository)
    {
        this.quotesRepository = quotesRepository;
    }

    public async Task<IEnumerable<Quote>> GetQuotes(int size)
    {
        var quotes = await quotesRepository.GetAll();
        var result = quotes.OrderBy(_ => Guid.NewGuid()).Take(size);
        return result;
    }

    public async Task<IEnumerable<Quote>> GetAllQuotes()
    {
        return await quotesRepository.GetAll();
    }

    public async Task<Quote?> GetRandomQuote()
    {
        var quotes = (await quotesRepository.GetAll()).ToList();
        if (quotes.Count == 0)
            return null;
        
        var ran = random.Next(0, quotes.Count);
        var result = quotes[ran];
        return result;
    }
}

File: KotlinApi/src/main/kotlin/no/dunder/mifflin/dundermifflinquotes/service/impl/QuotesService.kt
package no.dunder.mifflin.dundermifflinquotes.service.impl

import no.dunder.mifflin.dundermifflinquotes.db.IQuotesRepository
import no.dunder.mifflin.dundermifflinquotes.domain.dto.Quote
import no.dunder.mifflin.dundermifflinquotes.service.IQuotesService
import org.springframework.stereotype.Service
import kotlin.random.Random

@Service
class QuotesService(private val quotesRepository: IQuotesRepository) : IQuotesService {
    override fun getQuotes(size: Int): List<Quote> {
        val orderedQuotes = quotesRepository.getAll()
        val quotes = orderedQuotes.shuffled()

        if (size > quotes.size)
            return quotes

        return quotes.subList(0, size)
    }

    override fun getAllQuotes(): List<Quote> {
        return quotesRepository.getAll()
    }

    override fun getRandomQuote(): Quote? {
        val quotes = quotesRepository.getAll()

        if (quotes.isEmpty())
            return null

        return quotes[Random.nextInt(0, quotes.size)]
    }

}